### DISCUSSION TITLE
"context deadline exceeded"

### DISCUSSION THREAD

**Discussion by ebbek** [JanDeDobbeleer/oh-my-posh#5224] [Troubleshoot]
Hi

Environment: Powershell 7.4.3

When I run `oh-my-posh update` on an already updated OMP, I get the message in the subject. What does that mean?

<img width="365" alt="omp_message" src="https://github.com/JanDeDobbeleer/oh-my-posh/assets/38878653/d1fe92d3-9b9b-471f-90fb-bce7c5f39752">

**Comment by JanDeDobbeleer** [comment]
That we are unable to get information from GitHub in time. I will update the timeout.

**Reply by ebbek** [reply]
Thanks for the information.

I should add that I don't see this message in Ubuntu 24.04 running in WSL on the same machine, And this holds for both `bash` and `pwsh`.

**Comment by evilbaschdi** [comment]
any update on this? the problem persists!

**Reply by JanDeDobbeleer** [reply]
@evilbaschdi this is a client side issue, the timeout is 5 seconds already. Somehow GitHub isn't responding in a timely manner (could be proxy/Defender or some other configuration).

**Comment by evilbaschdi** [comment]
I understand! that could explain why I do not have that problem at the office where the internet connection is slightly faster (1000 Mbit/s
vs 250 Mbit/s).
But even having defender completly deactivated on windows 11 the problem still occurs. 
Also interessting: The problem does not exist on a Debian machine inside the same network or even on a WSL machine hostet on the affected windows 11 device.

Since I'm using DHCP that network settings _should_ be the same on each device. I do not have any proxy active.
Also I compared outputs of tracert / traceroute of windows and debian - no abnormalities.
The last things I've tried some minutes ago was to change the local DNS from my routers default IP to 8.8.8.8 and completly deactivated DNS over TLS inside my router. All without changes effect to the current situation.

What else could I try?

Edit:
Seems like the error occurs after 5 seconds. Do I have the possibility to raise that timeout on my own on my local machine?

**Reply by JanDeDobbeleer** [reply]
@evilbaschdi no, not yet. I'll see if I can up it. Although 5 seconds does imply it's not going to work when set to 10 either. It's a sub 1 second call under normal circumstances.

**Comment by mrwensveen** [comment]
I get this error when I'm on a VPN (possibly blocking github request or routing it through a very slow connection). If possible, check if you have a (slow) VPN running.

**Comment by znedw** [comment]
I've had this issue on one machine since switching on ipv6.... github doesn't support IPv6 and I think the timeout to fallback to IPv4 (happy eyeballs) can be more than 5 secs, so perhaps bumping it to 10 may allow enough time for the fallback...

**Reply by Bladehawke** [reply]
Can verify IPv6 as the culprit here as well

**Reply by soulpower11** [reply]
same, turned on IPv6 and got the error, turned it off and its fixed

**Comment by joaociocca** [comment]
this was srsly weird. I was having this same problem... then I read @JanDeDobbeleer saying that it's just a call to github, so it should be something of less than a second...

I decided to try a simple `curl 'https://api.github.com/repos/JanDeDobbeleer/oh-my-posh/releases/latest'`, and after it I tried the `update --force` again and OMP updated without any trouble...

**Comment by Zazucki** [comment]
This happens to me as well, but only on login to the server, it takes around 5 extra seconds before saying "❌ context deadline exceeded" and putting me into the shell. If I spawn a new bash shell after logging in, the time is minimal, around 1 sec or less, with no error.

**Reply by Zazucki** [reply]
After some testing, it looks like this was due to having "auto_upgrade = true" in my config file. Commenting this out solves the problem.

**Reply by ghost** [reply]
what file, bro? Please help me!

**Reply by chad3814** [reply]
> what file, bro? Please help me!

in your shell you have a line like:
`eval "$(oh-my-posh init zsh --config $HOME/.config/oh-my-posh/base.json)"`
your config file is at the end of that, in my case `$HOME/.config/oh-my-posh/base.json`

**Comment by chad3814** [comment]
So I was having this issue on my mac, which was getting an IPv6 address from my router, but I don't have an IPv6 uplink. Disabling getting an IPv6 address immediately solved the upgrading issue

**Comment by Nanome203** [comment]
You can try upgrading with winget

**Comment by sailro** [comment]
For me it is completely related to `ipv6`. See the following terminal:

![image](https://github.com/user-attachments/assets/1111a7f5-7a1e-43f8-b094-a89a3d5d2ee3)

- When I run `oh-my-posh upgrade` on my laptop using `wifi`/`ipv6`+`ipv4` it fails
- If I disable `ipv6` it works.
- If I re-enable `ipv6` it fails.

I don't know yet the root cause.

**Reply by JanDeDobbeleer** [reply]
The root cause is that github.com doesn't support IPv6
